src/Microsoft.AspNetCore.Mvc.Razor/Internal/MvcRazorDiagnosticSourceExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public static class MvcRazorDiagnosticSourceExtensions
+    {
+        public static void BeforeRazorView(
+            this DiagnosticSource diagnosticSource,
+            IRazorPage page,
+            ViewContext viewContext)
+        {
+            if (diagnosticSource.IsEnabled("Microsoft.AspNetCore.Mvc.Razor.BeforeRazorView"))
+            {
+                diagnosticSource.Write(
+                    "Microsoft.AspNetCore.Mvc.Razor.BeforeRazorView",
+                    new { page = page, viewContext = viewContext, });
```

Comment:
I tried to make it consistent with `BeforeView` and `AfterView` [here](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MvcViewFeaturesDiagnosticSourceExtensions.cs#L88).

Should these also include the `ActionDescriptor` and `HttpContext`?